protos/local/temporal/sdk/core/workflow_commands/workflow_commands.proto

```diff
@@ -309,3 +311,19 @@ message ModifyWorkflowProperties {
     // used as the value for the key being deleted.
     temporal.api.common.v1.Memo upserted_memo = 1;
 }
+
+// A reply to a `DoUpdate` job - lang must run the update's validator, and then immediately run
+// the handler, if the update was accepted.
+message UpdateResponse {
```

Comment:
Can you clarify, in the case of immediately completed update code, are you expecting two update responses, one for accepted and one for completed/rejected, or if you get just one do you assume completed/rejected implies accepted?

Reply:
Two. The comments below I was hoping clarified that, since both accepted and completed say they must be sent when the validator and handler pass, respectively.

LMK if you want me to change the comments

Reply:
Thanks, I was confused about failure, I'll ask on that field

Reply:
I see, I was confused by the term "rejected" being for update failure, will ask there.

Reply:
We could have separate commands, but either way you've got to send the expected sequence and this just has fewer thingies to deal with, so I figure let's stick with this unless there's some compelling reason.

As for the question - no, actually - you always send the same sequence. Check out the Rust SDK implementation as reference: https://github.com/temporalio/sdk-core/blob/8e0e0e010d6abd5b8c70830ed7a02243ebb2f1b0/sdk/src/workflow_future.rs#L245

Reply:
> As for the question - no, actually - you always send the same sequence

Ok, I think that's definitely worth comment update/clarification, because it says:

> if the update's validator has passed

but it may not even run. Maybe change that to:

> if the update's validator has passed or it was asked not to run

Then at the top of the proto or somewhere general add something like:

> If a `DoUpdate` is received, there must always immediately be an `UpdateResponse` with `accepted` or `rejected` set. `accepted` is validation success/skipped, `rejected` is validation failure. Only if the update was `accepted` can/should the update handler be run and any time later there can be _another_ `UpdateResponse` with either `completed` or `rejected` set depending on success/failure respectively.

Reply:
Sure, sounds good I'll add some extra info